### PR TITLE
(DOCSP-12860): customize mongodb docker image or container registry

### PR DESCRIPTION
**Pull request by jwilliams-mongo**
### All Submissions:

* [ x ] Have you opened an Issue before filing this PR?
* [ x ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ x ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ x ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

[JIRA](https://jira.mongodb.org/browse/DOCSP-12860)

Adds [section](https://github.com/mongodb/mongodb-kubernetes-operator/blob/e0df6ede9864253ede00da7242e2b035ce77ede2/docs/install-upgrade.md#configure-the-mongodb-docker-image-or-container-registry) on configuring the MongoDB Docker image and the container registry from which the Operator pulls this image.

### CODE REVIEW

File: docs/install-upgrade.md [hunk 1]
```diff
@@ -22,6 +23,7 @@ Before you install the MongoDB Community Kubernetes Operator, you must:
    git clone https://github.com/mongodb/mongodb-kubernetes-operator.git
    ```
 4. Review the possible Operator [deployment scopes](#understand-mongodb-community-operator-deployment-scopes) and configure the Operator to watch other namespaces, if necessary.
+5. Configure the [MongoDB Docker image or container registry](#configure-the-mongodb-docker-image-or-container-registry), if necessary.
```
Comment (melissamahoney-mongodb) [inline docs/install-upgrade.md:26]: Nit: Should we put the condition first? "If necessary, configure..." 
I know this messes up the parallelism of the list items a little bit. Up to you.

**Review by melissamahoney-mongodb — APPROVED** [review]
Nice! 🌮

File: docs/install-upgrade.md (hunk 1 above)
Comment (jwilliams-mongo) [reply docs/install-upgrade.md]: removed conditional statement. Made last two steps optional instead.

Comment (melissamahoney-mongodb) [reply docs/install-upgrade.md]: Nice, I like that better too.

**Review by chatton — APPROVED** [review]
LGTM! Thanks @jwilliams-mongo

**Pull request merged**